core/src/main/java/org/elasticsearch/index/query/support/InnerHitBuilder.java

```diff
@@ -202,7 +200,7 @@ public String getName() {
     }
 
     public InnerHitBuilder setName(String name) {
-        this.name = name;
+        this.name = Objects.requireNonNull(name);
```

Comment:
scratch that last line... the reason I didn't go with constructor arguments was because: the nested path / parent child type is known while parsing the xcontent for top level inner hits. After we have removed top level inner hits we can make this change easily.

(also name is just a label for reading back the inner hits in the response, lets say you have two nested queries for the same path that have inner hits enabled)